var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -249,3 +251,31 @@ def install(self, spec, prefix):
                 filter_file('mpcc_r)', 'mpcc_r mpifcc)', f, string=True)
                 filter_file('mpc++_r)', 'mpc++_r mpiFCC)', f, string=True)
                 filter_file('mpifc)', 'mpifc mpifrt)', f, string=True)
+
+    def _test_check_versions(self):
+        """Perform version checks on installed package binaries."""
+        spec_vers_str = 'version {0}'.format(self.spec.version)
+
+        checks = {
+            'ccmake': ([spec_vers_str], None),
+            'cmake':  ([spec_vers_str], None),
+            'cpack':  ([spec_vers_str], None),
+            'ctest':  ([spec_vers_str], None),
+        }
+
+        for exe in checks:
+            expected, status = checks[exe]
+            reason = 'test version of {0} is {1}'.format(exe, expected[0])
+            self.run_test(exe, ['--version'], expected, status, installed=True,
+                          purpose=reason, skip_missing=True)
+
+    def test(self):
+        """Perform smoke tests on the installed package."""
+        tty.warn('Expected results currently based on simple cmake builds')
+
+        if not self.spec.satisfies('@3.8.2:3.17.1'):
+            tty.warn('Expected results have not been confirmed for {0} {1}'
+                     .format(self.name, self.spec.version))
```

Comment:
Can we take out the warnings here? I think these would be better as a comment in the file than a user warning.

Reply:
Done.